### ISSUE TITLE
test_windowsErrorExcept fails on Windows with Python 3.5

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8776]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8776|
|Type|enhancement|
|Created|2016-08-20 23:54:02Z|
|Branch|https://github.com/twisted/twisted/tree/8776-rodrigc-windowserror-py3|

If I do:

```
python -m twisted.trial twisted.test.test_paths.ListingCompatibilityTests.test_windowsErrorExcept
```

I get:

```
twisted.test.test_paths
  ListingCompatibilityTests
    test_windowsErrorExcept ...                                          [FAIL]

===============================================================================
[FAIL]
Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\src\twisted\test\test_paths.py", line 277, in test_windowsErrorExcept
    self.assertRaises(filepath.UnlistableError, fwp.children)
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 414, in assertRaises
    return context._handle(lambda: f(*args, **kwargs))
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 308, in _handle
    self._returnValue = obj()
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 346, in __exit__
    self._expectedName, reason.getTraceback()),
twisted.trial.unittest.FailTest: builtins.OSError raised instead of UnlistableError:
 Traceback (most recent call last):
  File "C:\Users\Craig\twisted3\src\twisted\python\util.py", line 906, in runWithWarningsSuppressed
    return f(*args, **kwargs)
  File "C:\Users\Craig\twisted3\src\twisted\test\test_paths.py", line 277, in test_windowsErrorExcept
    self.assertRaises(filepath.UnlistableError, fwp.children)
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 414, in assertRaises
    return context._handle(lambda: f(*args, **kwargs))
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 308, in _handle
    self._returnValue = obj()
--- <exception caught here> ---
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 308, in _handle
    self._returnValue = obj()
  File "C:\Users\Craig\twisted3\src\twisted\trial\_synctest.py", line 414, in <lambda>
    return context._handle(lambda: f(*args, **kwargs))
  File "C:\Users\Craig\twisted3\src\twisted\python\filepath.py", line 332, in children
    subnames = self.listdir()
  File "C:\Users\Craig\twisted3\src\twisted\test\test_paths.py", line 262, in listdir
    "A directory's validness was called into question")
builtins.OSError: [Errno 267] A directory's validness was called into question


twisted.test.test_paths.ListingCompatibilityTests.test_windowsErrorExcept
-------------------------------------------------------------------------------
Ran 1 tests in 1.516s

FAILED (failures=1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__8776 8776
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8776_rodrigc_windowserror_py3 8776-rodrigc-windowserror-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471737242535733 1471737242535733
changetime__1471866530569890 1471866530569890
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/513

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

LGTM, merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset 9b3e4d219cd4da8d76fac9461b0ceec528ae5d58
```
#!CommitTicketReference repository="" revision="9b3e4d219cd4da8d76fac9461b0ceec528ae5d58"
Merge pull request #513 from twisted/8776-rodrigc-windowserror-py3

Author: rodrigc
Reviewer: Lukasa, acabhishek942
Fixes: #8776

Fix test_windowsErrorExcept on Windows with Python 3.5
```
